File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {

        int i = 0, j = numbers.size() - 1;
        vector<int> answer(2, -1);

        while (i <= j) {
            if (numbers[i] + numbers[j] == target) {
                answer[0] = i+1;
                answer[1] = j+1;
                return answer;
            } else if (numbers[i] + numbers[j] > target)
                j--;
             else
                i++;
        }
        return answer;
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {

        int[] answer=new int[2];

        int x=0, y=numbers.length-1;

        while(x<y)
        {
            if(numbers[x]+numbers[y]==target)
                {
                    answer[0]=x+1;
                    answer[1]=y+1;
                    return answer;
                }
                else if(target>numbers[x]+numbers[y])
                    x++;
                 else
                    y--;   
        }

        answer[0]=-1;
        answer[1]=-1;

        return answer;
    }
}